ndb/tests/unit/test_model.py

```diff
@@ -2722,6 +2656,55 @@ def test__put():
         with pytest.raises(NotImplementedError):
             entity._put()
 
+    @staticmethod
+    def test__lookup_model():
+        class ThisKind(model.Model):
+            pass
+
+        sentinel = object()
+        assert model.Model._lookup_model("ThisKind")
+        assert model.Model._lookup_model("NoKind", sentinel) is sentinel
+
+        with pytest.raises(model.KindError):
+            model.Model._lookup_model("NoKind")
+
+
+def test__entity_from_protobuf():
+    class ThisKind(model.Model):
+        a = model.IntegerProperty()
+        b = model.BooleanProperty()
+        c = model.PickleProperty()
+        d = model.StringProperty(repeated=True)
+        e = model.PickleProperty(repeated=True)
+        notaproperty = True
+
+    dill = {"sandwiches": ["turkey", "reuben"], "not_sandwiches": "tacos"}
+    gherkin = [{"a": {"b": "c"}, "d": 0}, [1, 2, 3], "himom"]
+    key = datastore.Key("ThisKind", 123, project="testing")
+    datastore_entity = datastore.Entity(key=key)
+    datastore_entity.update(
+        {
+            "a": 42,
+            "b": None,
+            "c": pickle.dumps(gherkin, pickle.HIGHEST_PROTOCOL),
```

Comment:
Why the use of `pickle.HIGHEST_PROTOCOL`? Isn't it the default?

Reply:
The default is `pickle.DEFAULT_PROTOCOL`. Sometimes they're the same, sometimes they're not, depending on Python version. `PickleProperty` explicitly uses `pickle.HIGHEST_PROTOCOL`, so this makes sure it's getting what it expects.

Reply:
~~~~
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pickle
>>> pickle.DEFAULT_PROTOCOL
3
>>> pickle.HIGHEST_PROTOCOL
4
~~~~